app/src/lib/git/branch.ts

```diff
@@ -94,41 +85,26 @@ export async function deleteBranch(
       `:${branch.nameWithoutRemote}`,
     ]
 
-    const opts = { env: envForAuthentication(account) }
-
     // If the user is not authenticated, the push is going to fail
     // Let this propagate and leave it to the caller to handle
-    await git(args, repository.path, 'deleteRemoteBranch', opts)
+    const result = await git(args, repository.path, 'deleteRemoteBranch', {
+      env: envForAuthentication(account),
+      expectedErrors: new Set<DugiteError>([DugiteError.BranchDeletionFailed]),
+    })
+
+    // It's possible that the delete failed because the ref has already
+    // been deleted on the remote. If we can identify that specific
```

Comment:
(non-blocking) small doc wording tweak to indicate that this is in fact what we're doing. for a minute i thought this was a speculative comment
```suggestion
    // been deleted on the remote. If we identify that specific
```